### PR TITLE
Exposed header date did change delegate method via day view delegate

### PR DESCRIPTION
**Pull request by danlozano**
Great framework!

Really saved my butt the other day where I had to implement a calendar in a day.

Realized I needed to know when a user clicks on a date, to do some logic. 

I figure this could be done in the Event Views for Date method. But I think it would be the wrong architecture, since that method is a data source and shouldn't be used as a delegate. Besides, if in the future the Events are gathered before a user actually clicks, for some sort of pre-fetching, then this would break that anyways.

### CODE REVIEW

**Comment by richardtop** [comment]
Please, take a look at #28, there is an interface example.
In your pull request an event would be called only if user changes date using header. If the date has been changed by swiping Timeline, the event won't be called.

**Comment by danlozano** [comment]
Uh uh!

Just realized that my hack/fix is not going to work in all cases! I should fix this for now in my local code 😂 

Yeah that API you suggested in #28 looks pretty good! 

Should I update my PR to implement that proposed API? Or is that something you are already working on?

Also, where should I take a look if I want to also handle the date changing by swiping the timeline? I need to fix this in my local code right now hehe

Thanks!!

**Comment by danlozano** [comment]
Ok!

So I updated the delegate methods to the ones you suggested in #28 .

Also, updated the location. Now they're called from the scrollViewDidScrollToViewAtIndex: method. Since that is called in both cases, when they scroll the timeline, and also when they click on a date in the header.

Not sure if that's how the WILL and DID are supposed to work. But could not think of another place to call the WILL from.

Comments appreciated :)

File: Source/DayView.swift [hunk 1]
```diff
@@ -129,7 +131,9 @@ extension DayView: PagingScrollViewDelegate {
   func scrollviewDidScrollToViewAtIndex(_ index: Int) {
     let timeline = timelinePager.reusableViews[index].timeline
     currentDate = timeline!.date
+    delegate?.dayView(dayView: self, willMoveTo: currentDate)
```
Comment (richardtop) [inline Source/DayView.swift]: I think, this method should be invoked before scrolling occurs. But I can fix it later.

File: Source/Timeline/EventView.swift [hunk 2]
```diff
@@ -29,6 +29,8 @@ open class EventView: UIView {
     }
   }
 
+  public var extraData: Any?
```
Comment (richardtop) [inline Source/Timeline/EventView.swift]: I'd call it `userInfo`, how about that?

Comment (danlozano) [inline Source/Timeline/EventView.swift]: userInfo makes sense!

Do you think it should also be a [String : Any] dictionary (as UserInfo usually is in Apple's frameworks) or leave it as an Any?

Comment (richardtop) [inline Source/Timeline/EventView.swift]: I think, `Any` is better:
* Allows user to specify own model class

Comment (danlozano) [inline Source/Timeline/EventView.swift]: cool

Comment (richardtop) [inline Source/Timeline/EventView.swift]: After this change, the PR would be ready to be merged.

**Comment by danlozano** [comment]
Cool! :)

**Comment by richardtop** [comment]
@danlozano your PR has just been released.
Sorry for delaying it, I was travelling.
Now you don't have to specify own fork when working with the library.

**Comment by danlozano** [comment]
@richardtop no worries, thanks! 👍 

Didn't quite get that last part. A pull request should be from my fork's master branch to yours, no? Or how are we supposed to do it now hehe.

Thanks again.

**Comment by richardtop** [comment]
@danlozano now you can specify `pod 'CalendarKit'` in your podfile.
Before, if you did so, CocoaPods was fetching the latest release, i.e. the one without your fix.
Now I've deployed a new release to trunk, so it will fetch the release with your fixes.

**Comment by danlozano** [comment]
OOOOH! Sorry I thought you were referring to something about pull requests.

Yeah got it, thanks again! 👍

**Pull request merged**